Fix `tool.tsh.common.TestKube/list_kube` flaky test

tigrato opened the pull request:
Tests under `tool.tsh.common.TestKube/list_kube` were picking the leftovers of another test which defined the `$HONE/.kube/config` file. This caused the test to be flaky because depending on the tests runnning order, some kube clusters appeared as selected when running `tsh kube ls`.

codingllama reviewed (approved):
Thanks for the quick fix!

codingllama commented:
Should we backport this too?

tigrato committed:
Fix `tool.tsh.common.TestKube/list_kube` flaky test

Tests under `tool.tsh.common.TestKube/list_kube` were picking the leftovers of another test which defined the `$HONE/.kube/config` file. This caused the test to be flaky because depending on the tests runnning order, some kube clusters appeared as selected when running `tsh kube ls`.

Signed-off-by: Tiago Silva <tiago.silva@goteleport.com>
